lib/dao/json_dao_container.es6.js

```diff
@@ -116,6 +123,19 @@ foam.CLASS({
         return this.StubFactorySingleton.create();
       },
     },
+    {
+      class: 'Function',
+      name: 'parseURL_',
+      transient: true,
+      factory: function() {
+        if (foam.isServer) {
+          const url = require('url');
+          return url.parse.bind(url);
+        } else {
+          return str => new URL(str);
+        }
```

Comment:
I'm sad about this code-smell - can you just verify that it's really needed?

Reply:
The underlying issue appears to be that, in NodeJS, `URL` isn't in the global namespace. We could use `new require('url').URL(...)` (or equivalent) instead of `require('url').parse(...)`, but the reference to the URL library doesn't go away. An alternative approach would be to have some compat code that always runs early that sets `URL = typeof URL === 'undefined' ? require('url').URL : URL`. I'm not sure that's much better, though.